### PR TITLE
Plot logicle fix

### PR DESCRIPTION
**Pull request by castillohair**
Solves #249.

### CODE REVIEW

**Comment by castillohair** [comment]
This pull request has code that improves function of the Logicle locator (the class that decides where to place the tickers). Previously, the locator could not handle cases in which only the "linear" region of the logicle scale was in screen. This led to failure when zooming into the linear region of a plot, and complete failure when opening a file in which negative events have a greater absolute value than positive events (because, in this case, all of the plot is in the "linear" region).

As for a7b39bf7db54f8fd5b296310d5caa79d335eab5a, the two files in #249 can now be plotted. The following are the density maps of forward/side scatter for both files:

File 1:
fail001

File 2:
fail002

To test this even further, I wrote the following code:

```python
import FlowCal as fc
import matplotlib.pyplot as plt

files = ['data_moake.fcs',
         'data_sea.fcs',
         'density_plot_fail.fcs']
channels = ['SSC', 'FSC-A', 'SSC-A']
xlims = [[None, [10, 200], [0, 200], [1e-3, 200],
          [0, 100], [0, 1], [0.02, 0.09], [50, 110]],
         [None, [-1e3, 2e5], [0, 2e5], [1e-3, 2e5],
          [-1e3, 1e3], [-4e3, -1e3], [-4e3, -990], [-4e3, 1e2]],
         [None, [-1e3, 1e4], [0, 1e4], [1e3, 1e4],
          [-2e4, -1e2], [-2e4, -5e3], [-2e4, 1e3], [1e5, 1e6]]]

for i, (file, channel, xlim) in enumerate(zip(files, channels, xlims)):
    print "File {}: {}...".format(i+1, file)
    d = fc.io.FCSData(file)
    d = fc.transform.to_rfi(d)

    plt.figure(figsize=(8, 10))
    for xi, xl in enumerate(xlim):
        print "   Zoom level {}...".format(xi + 1)
        plt.subplot(4, 2, xi + 1)
        fc.plot.hist1d(d, channel=channel)
        if xl is None:
            plt.title('Default')
        else:
            plt.title(str(xl))
        plt.xlim(xl)
    plt.tight_layout()
    plt.savefig('File{:03d}_zoom.png'.format(i+1), dpi=200)
    plt.close()
```

And tested it against [these files](https://github.com/taborlab/FlowCal/files/1082863/logicle_test.zip). I think this example code makes a good job a highlighting the improved abilities of the logicle locator, as well as its current limitations. Obviously, the code doesn't run without this fix.

For file 1 (Moake lab flow cytometer):
file001_zoom

Because of the decision to only place text under ticks that have an integral decade value (i.e. 10^a, where a is an integer), it is not hard to get into a situation like in subfigure 7, in which the zoom is such that no tick value is shown. This can also happen when using default matplotlib scaling, although it is a little harder to do. A quick solution from the user's perspective is to zoom out a little bit. Trying to place text on ticks with non-integral decade values can result in serious text agglomeration issues. I think fully resolving this would require me to implement a custom [Formatter](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/ticker.py), which is outside the scope of this pull request, and would require way more time that I have right now.

For file 2 (Moake lab's flow cytometer):
file002_zoom

Same issue in subplot 6. Again, the user can avoid this by zooming out a little (see subplot 7).

For file 3 (first file that failed to load in #249):
file003_zoom

In summary, this pull request completely solves issue #249. It seems that placement of text under ticks is not perfect, but it is not inferior to some native matplotlib scales, and therefore I will not be improving it further in this pull request. Nevertheless, this is a big improvement over the current `develop` branch.

**Review by JS3xton — APPROVED** [review]
I can't vouch for the technical details because I haven't taken the time to study the implementation closely, but the tests you've shown convince me that it's definitely better than it was. I also didn't see any glaring errors/typos from skimming the code.
